### PR TITLE
[FSTORE-1533] Importing hopsworks fails due to core dump while importing polars

### PR DESCRIPTION
**Pull request by manu-sj**
This PR fixes an issue in which hopworks is not importable because Polars throws a "" due to AVX instructions being unavailable.

https://github.com/pola-rs/polars/issues/2922 
https://github.com/pola-rs/polars?tab=readme-ov-file#legacy 

Fix Done: 
Make polars a optional dependency. The changes being done closely resembles the changes done for 4.0 (https://github.com/logicalclocks/feature-store-api/pull/1363)

JIRA Issue: https://hopsworks.atlassian.net/browse/FSTORE-1533

Related PRs: 
https://github.com/logicalclocks/hopsworks-api/pull/318
https://github.com/logicalclocks/loadtest/pull/454

**How Has This Been Tested?**

- [x] Unit Tests
- [ ] Integration Tests
- [x] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/core/vector_server.py [hunk 1]
```diff
@@ -59,6 +69,9 @@
 except ImportError:
     from avro.io import BinaryDecoder
 
+if HAS_POLARS or HAS_POLARS:
```
Comment (SirOibaf) [inline python/hsfs/core/vector_server.py]: Typo here.

File: python/hsfs/engine/python.py [hunk 2]
```diff
@@ -206,7 +218,6 @@ def _sql_offline(
         hive_config: Optional[Dict[str, Any]] = None,
         arrow_flight_config: Optional[Dict[str, Any]] = None,
     ) -> Union[pd.DataFrame, pl.DataFrame]:
```
Comment (SirOibaf) [inline python/hsfs/engine/python.py:220]: What about these annotations? Do they work if polars is not installed?

Comment (manu-sj) [reply python/hsfs/engine/python.py:220]: Yes there are just type hints are they would not cause any run time effects. I have tested it with the unit test and it was working. Still need to run some workflow tests and see if everything is fine.

https://stackoverflow.com/questions/41692473/does-python-type-hint-annotations-cause-some-run-time-effects

File: python/hsfs/core/vector_server.py (hunk 1 above)
Comment (manu-sj) [reply python/hsfs/core/vector_server.py]: Fixed type

**Review by SirOibaf — APPROVED** [review]

**Pull request merged**